pages/secure/sec_adcs_certificates.rst

```diff
@@ -0,0 +1,285 @@
+
+      certreq -new D:\secrets\graylog\hostname5-graylogreceiver.inf D:\secrets\graylog\hostname5-graylogreceiver.req
+      certreq -submit D:\secrets\graylog\hostname5-graylogreceiver.req
+
+This gives you a request ID, for example "531". Ask one of your PKI administrators to approve the request, for example::
+
+      certutil -resubmit 531
+
+Afterwards you can continue::
+
+      certreq -retrieve 531 D:\secrets\graylog\hostname5-graylogreceiver.cer
+      certreq -accept D:\secrets\graylog\hostname5-graylogreceiver.cer
+
+What all of this does is:
+
+#. Generate a keypair by your specifications.
+#. Generate a CSR for the keypair.
+#. Submit the CSR to the issuing CA.
+#. Approve the CSR on the issuing CA.
+#. Export the signed certificate from the issuing CA.
+#. Import the signed certificate into your current server's certificate store. 
+
+
+**SearchGuard admin**
+
+SearchGuard is used to add TLS/SSL encryption onto ElasticSearch for free. The product requires that the admin-user authenticates using a keypair and certificate. The generation process is similar to the one above, except that you won't be adding SANs, because the user does not have DNS names or IP addresses. The subject name will understandably also be different (e.g. *CN=searchguardadmin,OU=yourteam,O=yourorganization*), but be warned that it must match exactly with the account name in the SearchGuard configuration.
+
+
+Generating the keypair and certificates - conversion
+===============
+
+We showed earlier (in the table above) that each part of the Graylog stack has specific requirements for the format and files that are used to submit the keypair and the certificate. We will need to convert everything we have right now, in order to make them usable.
+
+
+**WARNING**
+
+Key materials are very sensitive information! You should not leave them lying around! Once you have finished the setup of all keys and certificates on the Graylog stack, you must delete all the files we've put into D:\secrets\graylog. Never leave key materials lying around!
+
+Also, please use strong passwords on all PFX and PKCS files! Store these passwords safely, in a password vaulting application.
+
+
+**CA Chain**
+
+Each application requires that you provide the CA chain of your PKI, for inclusion in its trust store. The following assumes that you have one root CA and one issuing CA and that you've put their respective certificates in D:\secrets\graylog::
+
+      openssl x509 -in rootca.crt -outform pem -out D:\secrets\graylog\rootca.pem
+      openssl x509 -in ca.crt -outform pem -out D:\secrets\graylog\ca.pem
+      type D:\secrets\graylog\rootca.pem > D:\secrets\graylog\cachain.pem
+      type D:\secrets\graylog\rootca.pem >> D:\secrets\graylog\cachain.pem
+
+The resulting cachain.pem file can be used in all Graylog stack applications for inclusion in the trust store. You will probably need to run the file through **dos2unix** first though, to fix line endings.
```

Comment:
Might want to expand this. If you will probably need to do it, the guide should cover how.

Reply:
```suggestion
The resulting cachain.pem file can be used in all Graylog stack applications for inclusion in the trust store. After uploading the file to Linux, you will need to run the file through **dos2unix** first, to fix line endings.
```